lib/postgrex/protocol.ex

```diff
@@ -176,6 +178,26 @@ defmodule Postgrex.Protocol do
     end
   end
 
+  @spec handle_execute(Postgrex.Stream.t, list, Keyword.t, state) ::
+    {:ok, Postgrex.Result.t, state} |
+    {:error, ArgumentError.t, state} |
+    {:error | :disconnect, Postgrex.Error.t, state}
+  def handle_execute( %Postgrex.Stream{ query: query } = stream, params, options, state ) do
+    %{buffer: buffer} = state
+    status = %{ notify: notify( options ), sync: :sync, prepare: :parse_execute }
+    case query_member?( state, query ) do
+      true ->
+        execute_send( state, status, stream, params, buffer )
+      false ->
+        case parse_describe_send( %{ state | buffer: nil }, %{ status | prepare: :parse_describe }, query, buffer ) do
```

Comment:
We can't call describe from `handle_execute`, we can use the same check as it won't call the describe protocol. The checks here are equivalent to the query ones. Consider making a helper function used by both.